theano/scan_module/scan_op.py

```diff
@@ -215,14 +215,37 @@ def make_node(self, *inputs):
                  " does not match the number of inputs given to scan.")
         new_inputs = [inputs[0]]
         # assert dtype is consistent
-        err_msg1 = ('When compiling the inner function of scan the '
-                    'following error has been encountered: The '
+        err_msg1 = ('When compiling the inner function of scan (the '
+                    'function called by scan in each of its iterations) '
+                    'the following error has been encountered: The '
                     '%s %s (argument number %d) has dtype '
-                    '%s and %d dimension(s). The corresponding slice %s '
-                    'however has dtype %s and %d dimension(s) (it should '
-                    'have the same dtype and one fewer dimensions). This '
-                    'should never happen, please '
-                    'report to theano-dev mailing list'
+                    '%s and %d dimension(s). The corresponding slice '
+                    'in the inner function of scan %s '
+                    'however has dtype %s and %d dimension(s). This '
+                    'slice in the inner function of scan should '
+                    'have the same dtype and one fewer dimension '
+                    'compared to its corresponding slice in the initial '
+                    'state (outputs_info in scan nomenclature). For example, '
+                    'if the inner function of scan returns a vector '
+                    '(dimensionality 1) of size d and scan uses the values of '
```

Comment:
Mentioning that a vector is dim1 and a matrix (below) is dim2 is probably overkill